### ISSUE TITLE
[bug] `following` endpoint is not returning correct urls of followed accounts

### ISSUE THREAD

**Issue by paulcuth** [superseriousbusiness/gotosocial#2786]
### Describe the bug with a clear and concise description of what the bug is.

GoToSocial's ActivityPub `following` endpoint is returning the account's own id (url) multiple times, instead of that of each account followed.

Example request (for who `@gotosocial@gts.superseriousbusiness.org` is following):
```
curl --header "accept: application/activity+json" --header "host: gts.superseriousbusiness.org" --header "date: Wed, 27 Mar 2024 22:15:11 GMT" --header 'signature: keyId="https://stars-81i.begin.app/users/bot#main-key",algorithm="rsa-sha256",headers="(request-target) host date",signature="Nkp+so76qXGOaNMConVmvCXixZNUe1I4ExHKpvkDG3hDfzpoMa9bnTyQnhY2X8YiIYSzbAu8NNEcGRYcQEFgV+3r7JbmRRafJnumWOgV97MyeYRn307g040rHv/P7KDlYwI6lry/epcuBCs3JxGVOOX5wkdr9d9RI4EnV3b7+zm8BHSISTtggJVWt477qNV8s/LZUQ2nKNXjgPQfvF1jFNVtgYvfGlWto3dK/cg4TyaI2/CfGpSjbLUECU4DQ+JD+yX/tLf8wOYInNu9zGXYge34FnFZ2I+IiahGHRlTr6ddMOLHdK8IGUa5bab0zZ/b63n+SULavFtjHd1kb5CvkA=="' https://gts.superseriousbusiness.org/users/gotosocial/following\?limit\=40
```

Response (formatted):
```
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "id": "https://gts.superseriousbusiness.org/users/gotosocial/following?limit=40",
  "next": "https://gts.superseriousbusiness.org/users/gotosocial/following?limit=40&max_id=014J79VNZH6JV9GPES87BD95R6",
  "orderedItems": [
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial",
    "https://gts.superseriousbusiness.org/users/gotosocial"
  ],
  "partOf": "https://gts.superseriousbusiness.org/users/gotosocial/following",
  "prev": "https://gts.superseriousbusiness.org/users/gotosocial/following?limit=40&since_id=01YW9GSVY6NX72D43EE624NXMP",
  "totalItems": 23,
  "type": "OrderedCollectionPage"
}
```

### What's your GoToSocial Version?

Whatever version is running on `gts.superseriousbusiness.org`

### What happened?

The `orderedItems` property in the response contains the account's own url multiple times.

Note: It seems to return the correct number of items in the page(s), it's just the value each of item that's incorrect.

### What you expected to happen?

The `orderedItems` field should contain different urls for each of the accounts followed by the subject account.

### How to reproduce it?

```
curl --header "accept: application/activity+json" --header "host: gts.superseriousbusiness.org" --header "date: Wed, 27 Mar 2024 22:15:11 GMT" --header 'signature: keyId="https://stars-81i.begin.app/users/bot#main-key",algorithm="rsa-sha256",headers="(request-target) host date",signature="Nkp+so76qXGOaNMConVmvCXixZNUe1I4ExHKpvkDG3hDfzpoMa9bnTyQnhY2X8YiIYSzbAu8NNEcGRYcQEFgV+3r7JbmRRafJnumWOgV97MyeYRn307g040rHv/P7KDlYwI6lry/epcuBCs3JxGVOOX5wkdr9d9RI4EnV3b7+zm8BHSISTtggJVWt477qNV8s/LZUQ2nKNXjgPQfvF1jFNVtgYvfGlWto3dK/cg4TyaI2/CfGpSjbLUECU4DQ+JD+yX/tLf8wOYInNu9zGXYge34FnFZ2I+IiahGHRlTr6ddMOLHdK8IGUa5bab0zZ/b63n+SULavFtjHd1kb5CvkA=="' https://gts.superseriousbusiness.org/users/gotosocial/following\?limit\=40
```

### Anything else we need to know?

I run [rss-is-dead.lol](https://rss-is-dead.lol), where this has been reported as an issue by a number of GoToSocial users on different servers.

As you can see by the [rss-is-dead.lol page for the account above](https://rss-is-dead.lol/user?profileUrl=https%3A%2F%2Fgts.superseriousbusiness.org%2Fusers%2Fgotosocial), it highlights this issue well.

**Comment by tsmethurst** [comment]
Oh weird bug :') OK thanks, will check it out.

**Comment by tsmethurst** [comment]
Fixed! Change will be included in 0.15.0.
